### PR TITLE
Add security headers with Helmet

### LINKED ISSUE #1120: Set our security headers using helmet
### Describe the ideal solution

Currently we set our NodeJS security headers ourselves, but ideally we want something cleaner like [helmet](https://github.com/helmetjs/helmet).

In our `web/scripts/serve.js`, we could easily replace our "custom middleware" and opt to use `helmet` if it can fulfill our needs.

### Describe your use cases

Code cleanup

### How frequently would you use such feature

N/A

### Describe alternatives you have considered

This doesn't add a new feature, it just helps with code cleanup

### References

https://github.com/helmetjs/helmet

## Acceptance Criteria

Modify our NodeJS Express server found under `web/scripts/serve.js` in order to use `helmet`. Specifically:
  - Make sure that no header is manually set, but we instead utilize helmet middlewares
  - Make sure that all current security headers are present after the migration
  - Make sure that all the app is functional after the migration and that no errors appear in the browser's console

### Additional context

When implementing this feature, be aware  that you'll need to update the dependenciess found in `deployments/Dockerfile` in `L62` to also include this new dep for  the  NodeJs server

### PR DESCRIPTION
**Pull request by kouknick**
## Description
Replace our "custom middleware" and use Helmet in order to set our security headers.

### Changes
New headers automatically added by Helmet:
- `X-Download-Options: noopen`
- `X-Permitted-Cross-Domain-Policies: none`

Modified headers:
- `X-XSS-Protection: 1; mode=block. `—> `X-XSS-Protection: 0`

Removed headers
- `X-Powered-by: N/A`

Closes #1120 

### Checklist:

- [x] Make sure you give an high-level overview of the problem & the solution
- [x] Make sure the related issue is referenced
- [x] Make sure screenshots are added if the UI is changed in any way
- [ ] Make sure all of the significant logic is covered by tests
- [x] Make sure your changes are accessible (addition of aria-attributes & correct HTML semantics)
- [x] Make sure you reference the related design document if it exists
- [ ] Make sure telemetry is added (analytics & errors)

### CODE REVIEW

**Comment by 3nvi** [comment]
@kouknick 

> X-XSS-Protection: 1; mode=block. —> X-XSS-Protection: 0

Where is the XSS protection header? Can't see it in the changes... Is it added automatically?

**Comment by kouknick** [comment]
@3nvi I couldn't remove all the custom headers as Helmet doesn't support Feature-Policy header. I am not really sure if we need this header because it is deprecated and will be replaced by [Permissions-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy) which is still not [supported](https://caniuse.com/permissions-policy) by browsers.

**Comment by kouknick** [comment]
> @kouknick
> 
> > X-XSS-Protection: 1; mode=block. —> X-XSS-Protection: 0
> 
> Where is the XSS protection header? Can't see it in the changes... Is it added automatically?

Correct

File: web/scripts/serve.js [hunk 1]
```diff
@@ -29,18 +30,32 @@ const app = express();
...
+      maxAge: 31536000,
+    },
+    dnsPrefetchControl: false,
+    expectCt: false,
+  })
+);
+
 app.use('*', (req, res, next) => {
-  res.header('X-Powered-by', 'N/A');
-  res.header('X-XSS-Protection', '1; mode=block');
```
Comment (3nvi) [inline web/scripts/serve.js:35]: Helmet's defaultt `'X-XSS-Protection'` is `0`. We want it to be `1` with a `block` mode

**Review by vorillaz — APPROVED** [review]

**Review by 3nvi — COMMENTED** [review]
Looks good. 1 small comment about XSS headers and we are good to go 🚢

File: web/scripts/serve.js (hunk 1 above)
Comment (kouknick) [reply web/scripts/serve.js:35]: I did some research online and I believe that is safer to set this header value to `0`. See [this](https://stackoverflow.com/questions/9090577/what-is-the-http-header-x-xss-protection/57802070#57802070) for details.
Helmet defaults `'X-XSS-Protection'` to `0` and has no available options in the documentation to change it to `1` with a `block` mode. If we still need it we have to apply it with a custom header.

Comment (3nvi) [reply web/scripts/serve.js:35]: Nope, it's fine. I've also verified that no external script gets blocked by any of our updated security headers.

**Review by 3nvi — APPROVED** [review]
LGTM @kouknick

**Pull request merged**